install/kubernetes/helm/istio/files/injection-template.yaml

```diff
@@ -207,6 +207,17 @@ containers:
     value: |
            {{ toJSON .ObjectMeta.Labels }}
   {{ end }}
+  {{- if .DeploymentMeta.Name }}
+  - name: ISTIO_META_WORKLOAD_NAME
+    value: {{ .DeploymentMeta.Name }}
+  {{ end }}
+  {{- if and .TypeMeta.APIVersion .DeploymentMeta.Name }}
+  - name: ISTIO_META_OWNER
+    value: kubernetes://api/{{ .TypeMeta.APIVersion }}/namespaces/{{ valueOrDefault .DeploymentMeta.Namespace `default` }}/{{ toLower .TypeMeta.Kind}}/{{ .DeploymentMeta.Name }}
```

Comment:
Keeping in mind that all this data is sent on every single ACK or DiscoveryRequest ( 100s of times ) to pilot - can we keep it smaller ? Do you really need kubernetes://api/... ? What exactly is needed in telemetry ? 
Namespace is already available, DeploymentMeta is passed just above.

Reply:
We _could_ required that the full owner field be recomputed by each individual user of the this metadata and only include the novel parts here, but it seems easier and safer to just precompute it once.

We can't change the owner values themselves, as that is an existing contract.